app/util/HelperFunctions.scala

```diff
@@ -14,4 +16,34 @@ object HelperFunctions {
     val t = AtomType.valueOf(atomType)
     Either.cond(t.isDefined, t.get, InvalidAtomTypeError)
   }
+}
+
+object Updater {
+  def update(map: Map[String, Any], path: String, value: JsLookupResult): Either[AtomAPIError, Map[String, Any]] =
+  try {
+    Right(updateNestedMap(map, path.split('.').toList, value.as[String]))
+  } catch {
+    case e: AtomAPIError => Left(e)
+  }
+
+  private def updateNestedMap[T](map: Map[String, Any], path: List[String], value: T): Map[String, Any] = path match {
+    case key :: Nil =>
+      // These checks shouldn't be necessary when using proper validation
+      if(map(key).isInstanceOf[Option[Any]]) {
```

Comment:
This is just a temporary check that looks at the value we want to update and if it is an option it will convert the new value to an option as well. Not doing this will result in replacing, for example, an `Option[Long]` with a `Long`. 

Ideally this check shouldn't be necessary anymore when we'll have some type checking.

This `updateNestedMap` method needs a lot of refactoring I'd say. But the changes depend a lot on how we want it to behave and there are things we haven't discussed yet. The way it is now represents just an example.